spec/services/preserved_object_handler_confirm_version_spec.rb

```diff
@@ -0,0 +1,335 @@
+  let(:ep) { Endpoint.find_by(storage_location: 'spec/fixtures/storage_root01/moab_storage_trunk') }
+  let(:pc) { PreservedCopy.find_by(preserved_object: po, endpoint: ep) }
+  let(:exp_msg_prefix) { "PreservedObjectHandler(#{druid}, #{incoming_version}, #{incoming_size}, #{ep})" }
+  let(:db_update_failed_prefix_regex_escaped) { Regexp.escape("#{exp_msg_prefix} db update failed") }
+  let(:po_handler) { described_class.new(druid, incoming_version, incoming_size, ep) }
+
+  describe '#confirm_version' do
+    it_behaves_like 'attributes validated', :confirm_version
+
+    context 'druid in db' do
+      before do
+        po = PreservedObject.create!(druid: druid, current_version: 2, preservation_policy: default_prez_policy)
+        PreservedCopy.create!(
+          preserved_object: po, # TODO: see if we got the preserved object that we expected
+          version: po.current_version,
+          size: 1,
+          endpoint: ep,
+          status: PreservedCopy::DEFAULT_STATUS
+        )
+      end
+
+      it 'stops processing if there is no PreservedCopy' do
+        druid = 'nd000lm0000'
+        diff_ep = Endpoint.create!(
+          endpoint_name: 'diff_endpoint',
+          endpoint_type: Endpoint.default_storage_root_endpoint_type,
+          endpoint_node: 'localhost',
+          storage_location: 'blah',
+          recovery_cost: 1
+        )
+        PreservedObject.create!(druid: druid, current_version: 2, preservation_policy: default_prez_policy)
+        po_handler = described_class.new(druid, 3, incoming_size, diff_ep)
+        results = po_handler.confirm_version
+        code = PreservedObjectHandlerResults::OBJECT_DOES_NOT_EXIST
+        exp_str = "ActiveRecord::RecordNotFound: Couldn't find PreservedCopy> db object does not exist"
+        expect(results).to include(a_hash_including(code => a_string_matching(exp_str)))
+        expect(PreservedObject.find_by(druid: druid).current_version).to eq 2
+      end
+
+      context "incoming and db versions match" do
+        let(:po_handler) { described_class.new(druid, 2, 1, ep) }
+        let(:exp_msg_prefix) { "PreservedObjectHandler(#{druid}, 2, 1, #{ep})" }
+        let(:version_matches_po_msg) { "#{exp_msg_prefix} incoming version (2) matches PreservedObject db version" }
+        let(:version_matches_pc_msg) { "#{exp_msg_prefix} incoming version (2) matches PreservedCopy db version" }
+        let(:updated_pc_db_msg) { "#{exp_msg_prefix} PreservedCopy db object updated" }
+
+        context 'PreservedCopy' do
+          context 'changed' do
+            it 'last_audited' do
+              orig = pc.last_audited
+              po_handler.confirm_version
+              expect(pc.reload.last_audited).not_to eq orig
+            end
+            it 'last_checked_on_storage' do
+              orig = pc.last_checked_on_storage
+              po_handler.confirm_version
+              expect(pc.reload.last_checked_on_storage).not_to eq orig
```

Comment:
i think i'd have expected `.to be > orig`, as in the subsequent tests (ditto the test before this one)

Reply:
this would be a good change, except in this test, the orig value is nil.

Reply:
yup, figured i was missing something!